src/globus_sdk/response.py

```diff
@@ -44,35 +29,40 @@ def __getitem__(self, key):
         try:
             return data[key]
         except TypeError:
-            logger.error(f"Can't index into responses of type {type(self)}")
-            # re-raise with an altered message -- the issue is that whatever
-            # type of GlobusResponse you're working with doesn't support
-            # indexing
+            log.error(
+                f"Can't index into responses with underlying data of type {type(data)}"
+            )
+            # re-raise with an altered message and error type -- the issue is that
+            # whatever data is in the response doesn't support indexing (e.g. a response
+            # that is just an integer, parsed as json)
+            #
             # "type" is ambiguous, but we don't know if it's the fault of the
             # class at large, or just a particular call's `data` property
-            raise TypeError(
-                "This type of GlobusResponse object does not support indexing."
-            )
+            raise ValueError("This type of response data does not support indexing.")
 
     def __contains__(self, item):
         """
-        ``x in GlobusResponse`` is an alias for ``x in GlobusResponse.data``
+        ``x in response`` is an alias for ``x in response.data``
         """
+        if self.data is None:
+            return False
         return item in self.data
 
-    @property
-    def data(self):
+    def get(self, key, default=None):
         """
-        Response data as a Python data structure. Usually a dict or
-        list.
+        ``get`` is just an alias for ``data.get(key, default)``, but with the added
+        check that if ``data`` is ``None``, it returns the default.
         """
-        return self._data
+        if self.data is None:
+            return default
+        # NB: `default` is provided as a positional because the native dict type
+        # doesn't recognize a keyword argument `default`
+        return self.data.get(key, default)
 
-    def get(self, *args, **kwargs):
-        """
-        ``GlobusResponse.get`` is just an alias for ``GlobusResponse.data.get``
-        """
-        return self.data.get(*args, **kwargs)
+    @property
+    @abc.abstractmethod
+    def data(self) -> Optional[Mapping]:
```

Comment:
I just noticed that this type annotation can be wrong in the case that a response contains JSON data which is not a dict. Maybe it should just be `Any`.

Some APIs return arrays (I think that should always, always, always be avoided, but my opinion doesn't change the reality).